packages/scoutgame-ui/src/components/claim/components/PointsClaimScreen/PointsClaimScreen.tsx

```diff
@@ -45,11 +64,66 @@ export function PointsClaimScreen({
     }
   };
 
+  async function handleWalletClaim() {
+    if (walletClient?.chain.id !== scoutProtocolChainId) {
+      await switchChainAsync({
+        chainId: scoutProtocolChainId,
+        addEthereumChainParameter: {
+          ...scoutProtocolChain,
+          rpcUrls: getChainById(scoutProtocolChainId)?.rpcUrls as string[]
+        }
+      });
+      return;
```

Comment:
this is the bug i posted - we should not return here. i think if u just await the promise it will wait til i switch wallets.